SWLOR.Game.Server/Native/ResolveAttackRoll.cs

```diff
@@ -301,35 +302,51 @@ private static void OnResolveAttackRoll(void* thisPtr, void* pTarget)
 
             var defenderWeapon = defender.m_pInventory.GetItemInSlot((uint)EquipmentSlot.RightHand);
             var defenderOffhand = defender.m_pInventory.GetItemInSlot((uint)EquipmentSlot.LeftHand);
-            bool saberBlock = false;
+            var saberBlock = false;
+            var shieldBlock = false;
+
+            var defenderPER = defender.m_pStats.GetDEXStat();
+            var defenderVIT = defender.m_pStats.GetCONStat();
 
             if(defenderWeapon != null && (
                 (BaseItem)defenderWeapon.m_nBaseItem == BaseItem.Lightsaber ||
                 (BaseItem)defenderWeapon.m_nBaseItem == BaseItem.Saberstaff))
             {
+                // Checking for saber reflect - they just need to be wielding a lightsaber or saberstaff.
                 for(var i = 0; i < defenderWeapon.m_lstPassiveProperties.Count; i++)
                 {
                     var ip = defenderWeapon.GetPassiveProperty(i);
-                    if (ip != null && ip.m_nPropertyName == (ushort)Core.NWScript.Enum.Item.ItemPropertyType.Light)
+                    if (ip != null && ip.m_nPropertyName == (ushort)ItemPropertyType.Light)
```

Comment:
I missed this the first time around. Why wouldn't we just check the base item type instead of the light property?

Reply:
If this has anything to do with lightsabers it is because the light property is the only consistent thing between them. 10ft = Padawan, 15ft = Knight and 20ft = Master.

Reply:
The rank doesn't matter based on the code though. It's only determining true lightsabers which is already determined above by the lightsaber/saberstaff check. I think this iteration over the properties to determine the light property is unnecessary and can be removed. Maybe the intention was to exclude electoblades, but those are their own item types which won't be picked up by the check on line 312/313.